### ISSUE TITLE
CREATE TABLE with enum types generates CHECK constraint rather than enum type reference in PostgreSQL

### ISSUE THREAD

**Issue by hbrackmann** [jOOQ/jOOQ#7597]
org.jooq.impl.Tools.toSQLDDLTypeDeclaration(Context<?>, DataType<?>) has been changed so that for Postgres a unappropriate default no creates a replacement for postgres enum types by mapping to varchar instead of using the original enum type.

```java
// [#5299] MySQL enum types


if (EnumType.class.isAssignableFrom(type.getType())) {
[…]

    switch (ctx.family()) {




        case MARIADB:
        case MYSQL: {
[…]

        default: {
====> NEW and producing unwanted results for Postgres
            type = emulateEnumType(enumType, enums);
            break;
        }
    }
}
```

### Versions:

- jOOQ: 3.11.0
- Java: 9

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.12. Backport scheduled for 3.11.2: #7606. This doesn't affect 3.10

**Comment by lukaseder** [comment]
This fix doesn't work correctly when trying to emulate MySQL enum types (not stored in a schema) in PostgreSQL, in case of which the `CHECK` constraint should be maintained.
